### PR TITLE
[#104] Support PolymorphicComponents as language pragma

### LINKED ISSUE #104: Support PolymorphicComponents
This is a minor issue, but when processing a couple of Haskell projects from GitHub I failed parsing a module from `haskell/parsec` because https://github.com/haskell/parsec/blob/783376aa67171130fe3e5d31deff150039615f35/src/Text/Parsec/Token.hs#L2 uses the language extension `PolymorphicComponents`. While https://kowainik.github.io/posts/extensions#stranger-things states that it is a Cabal synonym, it looks like GHC accepts this extension as well, although it is not listed on https://downloads.haskell.org/ghc/9.4.8/docs/users_guide/exts/table.html?highlight=extensions.

I am wondering, whether `PolymorphicComponents` should be defined as a synonym for `RankNTypes` in the extensions parser. If so, I am happy to create a small pull request.

PS: **Thanks very much** for creating this library in the first place. It helped me a lot!

### PR DESCRIPTION
**Pull request by jan-christiansen**
Resolves #104

### CODE REVIEW

**Comment by tomjaguarpaw** [comment]
Thanks!

File: extensions.cabal [hunk 1]
```diff
@@ -1,6 +1,6 @@
 cabal-version:       2.4
 name:                extensions
-version:             0.1.0.3
+version:             0.1.0.4
```
Comment (tomjaguarpaw) [inline extensions.cabal]: This is a behaviour change so needs a bump in the third component. But no need to bump here. I'll do it before I release.

```suggestion
```

**Comment by tomjaguarpaw** [comment]
Sorry, I made the wrong suggestion in my review comment. `version` needs to be reinstated, with the original version number.

**Comment by jan-christiansen** [comment]
I should probably have caught that.

Are there any guidelines on history rewriting in pull requests in this projects? That is, do you squash the pull request anyway, should I use force push to combine the commits into a single commit? Do you want to keep the history of the pull request?

**Comment by tomjaguarpaw** [comment]
That's no problem. Personally I prefer well-curated commits, which in this case means a squash (since it's so small). I don't mind if you do that or I do that.

**Review by tomjaguarpaw — APPROVED** [review]

**Comment by tomjaguarpaw** [comment]
Thanks!

**Comment by tomjaguarpaw** [comment]
Hi @jan-christiansen, you contributed this change but there's been no Hackage release incorporating it. Do you need one?

**Pull request merged**